lib/core/src/irods_service_account.cpp

```diff
@@ -0,0 +1,132 @@
+
+        if (!ret.ok()) {
+            // if config file not found, the machine probably does not have a server
+            return false;
+        }
+
+        std::ifstream svc_acct_cfg_strm{svc_acct_cfg_fn};
+        if (!svc_acct_cfg_strm.is_open()) {
+            // if we can't open the config file, it's likely due to a permission issue
+            // meaning it is extremely unlikely we are the service account
+            return false;
+        }
+
+        kvp_map_t service_acct_cfg;
+        uint_fast64_t cfg_line_no = 0;
+        for (std::string cfg_line; std::getline(svc_acct_cfg_strm, cfg_line);) {
+            cfg_line_no++;
+            boost::algorithm::trim(cfg_line); // trim whitespace (should handle CRLF)
+            if (cfg_line.empty()) {
+                // ignore empty lines
+                continue;
+            }
+            if (cfg_line.at(0) == '#') {
+                // ignore lines starting with #
+                continue;
+            }
+
+            // For now, let's just re-use our kvp-parsing functionality
+            // FIXME: does not handle spaces around "=", quotes, inline comments, or escape-newline
+            ret = parse_escaped_kvp_string(cfg_line, service_acct_cfg);
+            if (!ret.ok()) {
+                irods::log(LOG_WARNING,
+                           fmt::format("{} - {} parse failure on line {}",
+                                       __PRETTY_FUNCTION__,
+                                       SERVICE_ACCOUNT_CONFIG_FILE,
+                                       cfg_line_no));
+            }
+        }
+
+        const auto& svc_acct_itr = service_acct_cfg.find(SERTICE_ACCOUNT_KW);
+        if (svc_acct_itr == service_acct_cfg.end()) {
+            THROW(
+                KEY_NOT_FOUND, fmt::format("could not find {} in {}", SERTICE_ACCOUNT_KW, SERVICE_ACCOUNT_CONFIG_FILE));
+        }
+        service_account_username = svc_acct_itr->second;
+
+        // compare uids rather than usernames, to account for overlapping uids
+        // NOTE: do not reference the strings inside these structs, as their buffer goes out of scope
+        struct passwd svc_acct_pwd{};
+        struct passwd* svc_acct_pwdp = nullptr;
+        const auto _pwd_init_len = sysconf(_SC_GETPW_R_SIZE_MAX);
+        size_t pwd_buf_len = PWD_BUF_LEN_DEFAULT; // a reasonable buffer size in case sysconf gives us something awful
+        if (_pwd_init_len >= PWD_BUF_LEN_MINIMUM) {
+            pwd_buf_len = static_cast<size_t>(_pwd_init_len);
+        }
+        int pwd_e = 0;
+        while (true) {
+            // this sure is a lot of fuss to allocate a buffer for data we're not going to use anyway...
+            const std::unique_ptr<char[]> pwd_buf( // NOLINT(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+                new char[pwd_buf_len]);
+            pwd_e =
+                getpwnam_r(service_account_username.c_str(), &svc_acct_pwd, pwd_buf.get(), pwd_buf_len, &svc_acct_pwdp);
```

Comment:
this formatting looks off to me...

Reply:
This feels like a clang-format penalty break issue.

While the current formatting of that line isn't my ideal, I don't feel we should fight the formatter too much as there will always be things we don't like about it.

I'm fine with how it is now, but I'm also fine with you formatting that line differently and wrapping it in `clang-format off/on` comments if that improves readability.

Reply:
This is a `getpwnam_r` call we're talking about; it's pretty unreadable by default IMO. I'll leave it as-is.